CondFormats/HcalObjects/src/HcalCondObjectContainerBase.cc

```diff
@@ -9,16 +10,24 @@
 #include "FWCore/Utilities/interface/Exception.h"
 
 HcalCondObjectContainerBase::HcalCondObjectContainerBase(const HcalTopology* topo)
-    : packedIndexVersion_(0), topo_(topo) {
-  if (topo)
+    : packedIndexVersion_(0), topo_(topo), kSizeForDenseIndexing_(HcalZDCDetId::kSizeForDenseIndexingRun1) {
+  if (topo) {
     packedIndexVersion_ = topo->topoVersion();
+    HcalTopologyMode::Mode mode = topo->mode();
+    kSizeForDenseIndexing_ = (((mode == HcalTopologyMode::Run3) || (mode == HcalTopologyMode::Run4))
+                                  ? HcalZDCDetId::kSizeForDenseIndexingRun3
+                                  : HcalZDCDetId::kSizeForDenseIndexingRun1);
+  }
 }
 
 void HcalCondObjectContainerBase::setTopo(const HcalTopology* topo) {
   if ((topo) && (packedIndexVersion_ != 0) && (!topo->denseIdConsistent(packedIndexVersion_)))
     edm::LogError("HCAL") << std::string("Inconsistent dense packing between current topology (") << topo->topoVersion()
                           << ") and calibration object (" << packedIndexVersion_ << ")";
   topo_ = topo;
+  packedIndexVersion_ = topo_->topoVersion();
```

Comment:
>> Analyzing  src/CondFormats/HcalObjects/src/HcalLongRecoParam.cc 
src/CondFormats/HcalObjects/src/HcalCondObjectContainerBase.cc:28:25: warning: Called C++ object pointer is null [core.CallAndMessage]
   28 |   packedIndexVersion_ = topo_->topoVersion();
      |                         ^~~~~~~~~~~~~~~~~~~~
1 warning generated.